docs/DeveloperGuide.md

```diff
@@ -1,38 +1,462 @@
+<p align="center">
+Add Property Sequence Diagram
+</p>
+
+---
+
+### Delete Client/Property feature
+The **delete client/property** mechanism involves the following classes: ```ParseDeleteClient```,
+```ParseDeleteProperty```, ```CommandDeleteClient```, ```CommandDeleteProperty```,
+```ClientList```, ```PropertyList``` and ```PairingList```.
+
+Given below is an example usage scenario and how the delete client/property behaves at each step.
+
+**Step 1:** The user executes ```delete -client ic/INDEX``` or ```delete -property ip/INDEX```.
+The ```ParseDeleteClient``` or ```ParseDeleteProperty``` class is called respectively and the format of the user
+input is checked for any incorrect formatting.
+
+**Step 2:** If there are no errors, ```CommandDeleteClient``` or ```CommandDeleteProperty``` is called respectively.
+The ```CommandDeleteClient#execute()``` or ```CommandDeleteProperty#execute()``` method is then called.
+
+**Step 3:** The ```ClientList#deleteClient()``` or ```PropertyList#deleteProperty()``` method is called which 
+removes the Client or Property with that specific index from their respective ArrayList.
+
+**Step 4:** Any pairings involving that specific Client or Property is also deleted using the
+```pairingList#deletePairing()``` method. A message showing all the deleted pairs is shown to the user.
+
+**Step 5:** The corresponding line(s) in the respective files are deleted. The method is shown in the Storage
+Implementation section.
+
+The following *class diagram* shows all the classes involved in the **delete client/property** operation
+and their relationships.
+
+![Delete Client/Property Class Diagram](diagrams/DeleteClientPropertyCD.png)
+
+The following *sequence diagram* shows how the **delete client** operation works, showcasing the
+```ClientList#deleteClient()``` method.
+
+![Delete Client Sequence Diagram](diagrams/DeleteClientSD.png)
+
+The following *sequence diagram* shows how the **delete property** operation works, showcasing the
+```PropertyList#deleteProperty()``` method.
+
+![Delete Property Sequence Diagram](diagrams/DeletePropertySD.png)
+
+---
+
+### PairingList
+
+```PairingList``` facilities that pair and unpair commands by storing client-property pairs.
+
+When client rents a property, the client and property form a pair.
+
+*  ```PairingList``` uses a hash map to represent these client-property pairs, where the key is a ```Client``` object
+  and the value is a ```Property``` object.
+* A hash map is chosen due to its constant time lookup performance, making it efficient at querying the property that a
+  client is renting.
+* Also, the Java HashMap prevents duplicate keys, which dovetails nicely with the fact that real-life tenants only have
+  one place of residence at any time.
+
+#### Pair
+
+The sequence diagram for the pair command is called is shown below:
+
+![PairingList Add Pair Sequence Diagram](diagrams/PairingListAddPairSD.png)
```

Comment:
This sequence diagram attempts to explain too much. In fact, sequence diagrams should only be used to aid the explanation of concepts difficult to grasp, and not depict the entire program function. Suggest using just a small portion where program sequence is not simply linear.